lib/common.js

```diff
@@ -0,0 +1,183 @@
+export const defaultLocale = 'en';
+export const supportedBaseLocales = ['ar', 'da', 'de', 'en', 'es', 'fr', 'ja', 'ko', 'nl', 'pt', 'sv', 'tr', 'zh'];
+export const supportedLocales = [
+	'ar-sa',
+	'da-dk',
+	'de-de',
+	'en-us',
+	'en-ca',
+	'en-gb',
+	'es-mx',
+	'fr-ca',
+	'fr-fr',
+	'ja-jp',
+	'ko-kr',
+	'nl-nl',
+	'pt-br',
+	'sv-se',
+	'tr-tr',
+	'zh-cn',
+	'zh-tw'
+];
+
+function tryResolve(langTag) {
+
+	if (langTag === undefined || langTag === null) return null;
+
+	if (supportedLocales.indexOf(langTag) > -1) {
+		return langTag;
+	}
+
+	const subtags = langTag.split('-');
+	if (subtags.length < 2) {
+		if (supportedBaseLocales.indexOf(langTag) > -1) {
+			return langTag;
+		}
+		return null;
+	}
+
+	if (supportedBaseLocales.indexOf(subtags[0]) > -1) {
+		return subtags[0];
+	}
+
+	return null;
+
+}
+
+export function getLanguage() {
+	const settings = getDocumentLocaleSettings();
+	const langTag = tryResolve(settings.language) ||
+		tryResolve(settings.fallbackLanguage) || defaultLocale;
+	return langTag;
+}
+
+export function merge(obj1, obj2) {
+	if (obj2 === undefined || obj2 === null || typeof(obj2) !== 'object') {
+		return;
+	}
+	for (var i in obj2) {
+		// eslint-disable-next-line no-prototype-builtins
+		if (obj1.hasOwnProperty(i)) {
+			if (typeof(obj2[i]) === 'object' && typeof(obj1[i]) === 'object') {
+				merge(obj1[i], obj2[i]);
+			} else {
+				obj1[i] = obj2[i];
+			}
+		}
+	}
+}
+
+export function validateFormatValue(value) {
+	if (value === undefined || value === null) {
+		return 0;
+	}
+	if (typeof value === 'string') {
+		value = parseFloat(value);
+	}
+	if (isNaN(value) || typeof value !== 'number') {
+		throw new RangeError('value is out of range.');
+	}
+	return value;
+}
+
+class DocumentLocaleSettings {
+
+	constructor() {
+		this._htmlElem = window.document.getElementsByTagName('html')[0];
+		this._listeners = [];
+		this._observer = new MutationObserver(this._handleObserverChange.bind(this));
+		this._observer.observe(this._htmlElem, { attributes: true });
+		this.sync();
+	}
+
+	get language() { return this._language; }
+	set language(val) {
+		const normalized = this._normalize(val);
+		if (normalized === this._language) return;
+		this._language = normalized;
+		this._listeners.forEach((cb) => cb());
+	}
+
+	get fallbackLanguage() { return this._fallbackLanguage; }
+	set fallbackLanguage(val) {
+		const normalized = this._normalize(val);
+		if (normalized === this._fallbackLanguage) return;
+		this._fallbackLanguage = this._normalize(val);
```

Comment:
nit: could just use `this._fallbackLanguage = normalized` like in `set language`